sycl/test-e2e/ESIMD/unified_memory_api/Inputs/block_store.hpp

```diff
@@ -97,81 +117,260 @@ bool testUSM(queue Q, uint32_t Groups, uint32_t Threads,
+
+  if constexpr (TestPVCFeatures) {
+    // Using cache hints adds the requirement to run tests on PVC.
+    // Also, PVC variant currently requires a) power-or-two elements,
+    // b) the number of bytes stored per call must not exceed 512,
+    // c) the alignment of USM ptr + offset to be 4 or 8-bytes(for 8-byte
+    // element vectors).
+    constexpr size_t RequiredAlignment = sizeof(T) <= 4 ? 4 : 8;
+    properties PVCProps{cache_hint_L1<cache_hint::write_back>,
+                        cache_hint_L2<cache_hint::write_back>,
+                        alignment<RequiredAlignment>};
+    // Only d/q-words are supported now.
+    // Thus we use this I32Factor for testing purposes and convenience.
+    constexpr int I32Factor =
+        std::max(static_cast<int>(sizeof(int) / sizeof(T)), 1);
+
+    Passed &= testUSM<T, 1 * I32Factor, !CheckMask, CheckProperties>(Q, 2, 4,
+                                                                     PVCProps);
+    Passed &= testUSM<T, 2 * I32Factor, !CheckMask, CheckProperties>(Q, 5, 5,
+                                                                     PVCProps);
+    Passed &= testUSM<T, 4 * I32Factor, !CheckMask, CheckProperties>(Q, 5, 5,
+                                                                     PVCProps);
+    Passed &= testUSM<T, 8 * I32Factor, !CheckMask, CheckProperties>(Q, 5, 5,
+                                                                     PVCProps);
+    Passed &= testUSM<T, 16 * I32Factor, CheckMask, CheckProperties>(Q, 5, 5,
+                                                                     PVCProps);
+    Passed &= testUSM<T, 32 * I32Factor, !CheckMask, CheckProperties>(Q, 2, 4,
+                                                                      PVCProps);
+
+    // This call (potentially) and the next call (guaranteed) store the biggest
+    // store-able chunk, which requires storing with 8-byte elements, which
+    // requires the alignment to be 8-bytes or more.
+    properties PVCAlign8Props{cache_hint_L1<cache_hint::write_back>,
+                              cache_hint_L2<cache_hint::write_back>,
+                              alignment<8>};
+    Passed &= testUSM<T, 64 * I32Factor, !CheckMask, CheckProperties>(
+        Q, 7, 1, PVCAlign8Props);
+    if constexpr (sizeof(T) <= 4)
+      Passed &= testUSM<T, 128 * I32Factor, CheckMask, CheckProperties>(
+          Q, 1, 4, PVCAlign8Props);
+
+  } // TestPVCFeatures
+
+  return Passed;
+}
+
+template <typename T, bool TestPVCFeatures> bool test_block_store_acc(queue Q) {
+  constexpr bool CheckMask = true;
+  constexpr bool CheckProperties = true;
+  properties Align16Props{alignment<16>};
+  properties AlignElemProps{alignment<sizeof(T)>};
+
+  bool Passed = true;
+
+  // Test block_store() that is available on Gen12 and PVC.
+
+  if constexpr (sizeof(T) >= 4)
+    Passed &=
+        testACC<T, 4, !CheckMask, CheckProperties>(Q, 2, 4, AlignElemProps);
```

Comment:
I think the usage of 'AlignElemProps' is user's error here. The test works because actual alignment is 16: sizeof(T) >=4 and N=4 gives 16 or more bytes.

If user cannot guarantee the byte offsets to be 16-byte aligned, then the only option for us is to generate LSC call, but this part of the test is supposed to pass on Gen12.

Same for the next call.

Reply:
I had in mind to make these tests more strict by doing something like:
If user says the alignment is 4-bytes, then intentionally manage the offset to be 4-bytes aligned and not 16-bytes aligned.
So, for these tests it would mean allocating a bit bigger buffers and add start reading with extra offset=4 (+4 to some 16-byte aligned boundary).

Reply:
I changed the Gen12 ones to just use 16-byte alignment to not complicate the test even more